audio/filter/af_scaletempo2_internals.c

```diff
@@ -524,10 +512,8 @@ int mp_scaletempo2_fill_input_buffer(struct mp_scaletempo2 *p,
         return 0;
 
     int required_size = read + p->input_buffer_frames;
-    if (required_size > p->input_buffer_size)
-        resize_input_buffer(p, required_size);
-
     for (int i = 0; i < p->channels; ++i) {
+        MP_TARRAY_GROW(p, p->input_buffer[i], required_size);
```

Comment:
```suggestion
        MP_TARRAY_GROW(p, p->input_buffer[i], required_size - 1);
```

Reply:
variable renamed to `last_index` here too